apps/system/utils/security_level_cmd.c

```diff
@@ -0,0 +1,127 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/ss_slot_index.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/prctl.h>
+#include <apps/shell/tash.h>
+#include <security/security_ss.h>
+
+/********************************************************************************
+ * Private Functions
+ ********************************************************************************/
+static void show_usage(void) 
+{
+	printf("\nUsage : sec_level [OPTION]...\n");
+	printf("Set your security level high or low, or get the current security level status.\n\n");
+	printf("Options :\n");
+	printf("high		set security level 1(high)\n");
+	printf("		Your security level becomes high\n\n");
+	printf("low		set security level 0(low)\n");
+	printf("		Your security level becomes low\n\n");
+	printf("status		show current value of security level\n\n");
+}
+
+static void tash_security_level(int argc, char **args)
+{
+        security_handle hnd;
+	security_data ss_data = {NULL, 0};
+	security_error ret;
+
+	if (argc < 2) {
+		printf("Use [--help] option if you want to know how to use.\n");
+		return;
+	}
+
+	/* Show usage */
+	if (!strncmp(args[1], "--help", 7)) {
+		show_usage();
+		return;
+	}
+
+	/* Init security handler */
+	ret = security_init(&hnd);
```

Comment:
The codes of about seurity_level setting and reading is better to be gathered in one place.

How about move the this code to os/kernel/security_level and just all here?

Please let me know your opinion.

Reply:
Please confirm this. Is that your opinion?

1. When user change value in TASH, security_level_cmd.c change the `security_level` variable, which is defined in security_level.c
2. In security_level.c, write `security_level` variable in secure storage.

Reply:
yes I think this

Reply:
To change the code according to your opinion, we must create another API to access `security_level` or define `security_level` as a global variable. It means that other apps can also change this variable.
However, `security_level` will only be used in kernel, and there is no reason to access it from other apps.
So I think the current condition is a better way.

Reply:
I agree that a new API should be added. However, security_level does not need to be a global variable.

Is security_level.c an API for reading and writing only security_level variables?

The security_level variable is changed to the secure_storage value.
However, the code that actually write secure_storage value does not exist in security_level.c.
We do not know who set the secure_storage to set the security_level to on/off in security_level.c.

but It is just my opinion.